src/libstd/io/impls.rs

```diff
@@ -363,6 +375,26 @@ impl Write for Vec<u8> {
     }
 }
 
+fn write_vec_oom(vec: &mut Vec<u8>, buf: &[u8]) -> io::Result<usize> {
```

Comment:
Also a minor style question: should I call this function `write_vec_oom` or `vec_write_oom`?